### PR TITLE
xcursor: new module for configuring X cursor theme

### PR DESCRIPTION
**Pull request by league**
Addresses #200. Seems to work just fine. Should I add news item?

### CODE REVIEW

File: modules/xcursor.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    };
+  };
+
+in
+
+{
+  meta.maintainers = [ maintainers.league ];
+
+  options = {
+    xcursor = mkOption {
```
Comment (rycee) [inline modules/xcursor.nix]: I'd suggest the name `xsession.cursor` for this option.

File: modules/xcursor.nix [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      type = types.nullOr cursorType;
+      default = null;
+      description = ''
+        The X cursor theme and settings. There package pkgs.xorg.xcursorthemes
```
Comment (rycee) [inline modules/xcursor.nix]: I'd change `There package pkgs.xorg.xcursorthemes` to `The package <varname>xorg.xcursorthemes</varname>`. I.e., using removing the `pkgs.` and using docbook tags to indicate that this is a variable name.

File: modules/xcursor.nix [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      gtk-cursor-theme-name="${cfg.name}"
+      gtk-cursor-theme-size=${toString cfg.size}
+    '';
+
+    gtk.gtk3.extraConfig = {
+      "gtk-cursor-theme-name" = cfg.name;
+      "gtk-cursor-theme-size" = cfg.size;
+    };
+
+
```
Comment (rycee) [inline modules/xcursor.nix]: Can remove this stray line.

File: modules/xcursor.nix (hunk 1 above)
Comment (rycee) [reply modules/xcursor.nix]: And yeah, I think my choice of name for the `xresources` option was flawed. I think it should be called something like `xsession.resourceProperties` instead 😄

Comment (rycee) [reply modules/xcursor.nix]: Actually, reading the xsetroot man page it talks about "pointer cursor" so the option name `xsession.pointerCursor` might be more descriptive.

**Comment by rycee** [comment]
Yeah, this looks pretty much exactly how I would write it so I approve ;-) I just made some minor comments and I do think the `xsession.pointerCursor` name is better.

**Comment by league** [comment]
Thanks, I think I applied all the suggestions! This became four commits, but IIRC you can squash automatically while merging PR? If not I could force-push the squashed version instead.

**Comment by league** [comment]
Crud hang on, I broke news in the merge?

**Comment by league** [comment]
Yup, should be okay now.

**Comment by rycee** [comment]
Looks and works great! Thanks! Now I also enjoy a cursor I actually can see :-)

I've squashed and rebased this PR into master in 563a20fc82124abebd75a1fbaa6b6ead835d2553.

**Comment by league** [comment]
@rycee I wasn't kidding… life-changing, right? :)

**Pull request closed without merging**